### ISSUE TITLE
Not Autogenerating for Remote Routes.

### ISSUE THREAD

**Issue by nidhhoggr** [strongloop/loopback-connector-remote#52]
I tried asking on the gitter channel but go no response or acknowledgement so here I am.

The documentation states that the connector will automatically generate functions to make rest calls for remote endpoints that are publicly exposed. This doesn't seem to be the case because when I console log the remote data source models property I am only getting the models of the local API. Also in the loopback connector example it shows that not only do you need to make sure that the models are public exposed on the remote API but you also have to establish them as models on the local API. If this is the case than how would I connect to a remote endpoint that has the same name as a model on my existing API? So two things to confirm here:

1) Remote Data source is not automatically generating an API for the remote endpoint.
2) How should I connect to a remote endpoint that has the same name as a model on a local endpoint?

**Comment by nidhhoggr** [comment]
After further research and digging into the code the compiler in the loopback-boot node module does not allow models with the same name from different data sources. This is because the data structure storing the models is using the model name as the key. 

Also the loopback-remote-connector is only considering models that are remote and are specified in the model-config.json which entails having an associated model definition and model file.

**Comment by nidhhoggr** [comment]
Here is how I solved the issue but I think its super hacky yet works nonetheless.

Two APIs 
 - Offer API
 - Vendor API

Both API's have a model named offer
 - Offer

These models needs to be able to synch from Offer API to Vendor API. With the current model registry bootloading implementing have two datasources with the same model name communicate to eacother is no possible. The following is the solution

**Offer Api**

- server/model-config.json
```json
  ...
  "vendor_offer": {
    "dataSource": "vendor-api",
    "public": false
  },
```
- server/datasources.json
```json
  ...
  "vendor-api": {
    "url": "http://0.0.0.0:3003/api",
    "name": "vendor-api",
    "connector": "remote"
  }
```
- common/models/vendor_offer.json 
```json
{
  "name": "vendor_offer"
}
```
- server/datasources.json
```json
  ...
  "postgresDs": {
    "host": "localhost",
    "port": 5432,
    "database": "provider_api",
    "username": "postgres",
    "password": "postgres",
    "name": "postgresDs",
    "connector": "postgresql",
    "debug": true,
    "ssl": false
  },
```
**Vendor Api**

- server/model-config.json
```json
  ...
  "vendor_offer": {
    "dataSource": "postgresDs",
    "public": true
  },
```



- server/vendor-offer.json
```json
{
  "name": "vendor_offer",
  "base": "offer",
  "options": {
    "postgresql": {
      "table":"offer"
    }
  }
}
```

In conclusion, I have to create model on the remote API just for the purposes of acting as a proxy for the intended model because the model cannot have the same name as a any other model in the remote API.

**Comment by nidhhoggr** [comment]
Found a solution yielding results identical to that above requiring configuration on the local API only.

**Offer Api**

- server/model-config.json
```json
  ...
  "vendor_offer": {
    "dataSource": "vendor-api",
    "public": false
  },
```
- server/datasources.json
```json
  ...
  "vendor-api": {
    "url": "http://0.0.0.0:3003/api",
    "name": "vendor-api",
    "connector": "remote"
  }
```
- common/models/vendor_offer.json 
```json
{
  "name": "vendor_offer",
  "http": {"path":"/offers"}
}
```

NOTE: this works in the code but is not reliable in the explorer.  The explorer seems to clash with the local definition of offers which is a result of changing would be available `/api/vendor_offers` to `/api/offers`. The explorer will also have two indistginguishable `/api/offers` endpoints available.

**Resolution** Before closing this issue I would recommend updating the README to contain additional vital information in making the remote endpoint available (edits to local model-config.js file) and revising wording that only the remote endpoints specified in the model-config are 'automatically generated'

**Comment by mrfelton** [comment]
Same issue here. The docs in the README don't appear to match what is actually required to use the remote connector.
